api/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -62,6 +62,22 @@ public abstract void applyRequestMetadata(
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1914")
   public abstract void thisUsesUnstableApi();
 
+  /**
+   * Check if the security level on the transport is higher than or equal to the minimum security
+   * level required by {@link CallCredentials} in order to transfer it.
+   *
+   * <p>It should be called in {@link applyRequestMetadata} before sending any
+   * individual RPC. The RPC should not be sent if the API returns false.
+   * More details can be found in https://github.com/grpc/proposal/pull/167.
+   *
+   * @param requestInfo request-related information
+   * @param minSecurity minimum security level required by {@link CallCredentials}
+   */
+  public final boolean checkSecurityLevel(RequestInfo requestInfo, SecurityLevel minSecurity) {
```

Comment:
`check` makes it seem like this will throw. Maybe name it `allowedSecurityLevel` or `fulfullsSecurityLevel` or similar that makes it more obvious it returns a boolean?

CC @jiangtaoli2016